sssd_test_framework/utils/sssd.py

```diff
@@ -783,3 +784,50 @@ def autofs(self) -> None:
         """
         self.sssd.authselect.select("sssd")
         self.sssd.enable_responder("autofs")
+
+    def proxy(
+        self,
+        proxy: Literal["files", "ldap"],
+        provider: str | list[str] = "id",
+        proxy_file: str = """""",
+        sever_hostname: str = """""",
+        domain: str | None = None,
+    ):
+        """
+        :param proxy: ``ldap`` or ``files``
+        :type proxy: str
+        :param provider: SSSD providers (id, auth, chpass, ...)
+        :type provider: str | list[str]
+        :param proxy_file: Config for proxy
+        :type proxy_file: str
+        :param sever_hostname: LDAP (LDAP server hostname)
+        :type sever_hostname: str
+        :param domain: Existing domain name, defaults to None (= default domain)
+        :type domain: str | None, optional
+        """
+        if domain is None:
+            domain = self.sssd.default_domain
+
+        match proxy:
+            case "ldap":
+                target = "sssdproxyldap"
+                self.sssd.fs.write(f"/etc/pam.d/{target}", proxy_file)
+                self.sssd.fs.write(
+                    "/etc/nslcd.conf", f"uid nslcd\ngid ldap\nuri " f"ldap://{sever_hostname}\n", dedent=False
+                )
+                self.sssd.svc.restart("nslcd")
+            case "files":
```

Comment:
I don't think this is correct and it's confusing by the naming. Proxy ldap forwards to an PAM stack while files doesn't, but it's called files. 

So in case "files": 
[domain/files]
id_provider = files

is all that is required. 

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system-level_authentication_guide/configuring_domains#configuring-the-files-provider-for-sssd